src/typing/type_inference_js.ml

```diff
@@ -1220,46 +1220,53 @@ and statement_decl cx = Ast.Statement.(
           in
           Env_js.init_var cx local_name env_entry
         | None -> (
+          (* Parser sanity check *)
           match specifier with
-          | Some(ImportDeclaration.Named(_, named_specifiers)) ->
-            let init_specifier (specifier_loc, specifier) = (
-              let (loc, remote_ident) =
-                specifier.ImportDeclaration.NamedSpecifier.id
-              in
-              let remote_name = remote_ident.Ast.Identifier.name in
-              let (local_name, reason) = (
-                match specifier.ImportDeclaration.NamedSpecifier.name with
-                | Some(_, { Ast.Identifier.name = local_name; _; }) ->
-                  let reason_str =
-                    spf "%s { %s as %s }" import_str remote_name local_name
-                  in
-                  (local_name, (mk_reason reason_str loc))
-                | None ->
-                  let reason_str = spf "%s { %s }" import_str remote_name in
-                  (remote_name, (mk_reason reason_str loc))
-              ) in
-              let tvar = Flow_js.mk_tvar cx reason in
-              let env_entry =
-                Scope.create_entry ~for_type:isType tvar tvar (Some specifier_loc)
-              in
-              Env_js.init_var cx local_name env_entry;
-            ) in
-            List.iter init_specifier named_specifiers
-          | Some(ImportDeclaration.NameSpace(_, (loc, local_ident))) ->
-            let local_name = local_ident.Ast.Identifier.name in
-            let reason =
-              mk_reason (spf "%s * as %s" import_str local_name) loc
+          | None ->
+            failwith (
+              "Parser error: Non-default imports must have a set of " ^
+              "named import specifiers!"
```

Comment:
Isn't it possible to import for side effects only, without specifying any bindings, default or non? See #574 for an issue, and [the spec at 15.2.2; `import ModuleSpecifier`](http://www.ecma-international.org/ecma-262/6.0/#sec-imports).